tbot/machine/linux/path.py

```diff
@@ -71,13 +61,152 @@ def __init__(self, host: H, *args: typing.Any) -> None:
         return self._host
 
+    def _local_str(self) -> str:
+        return str(self._path)
+
+    def at_host(self, host: H) -> str:
+        """
+        Convert this ``Path`` into a string representation assuming it should
+        be valid for the machine ``host``.  An exception is raised if this
+        ``Path`` is for a different machine instead.
+
+        This prevents accidentally using a ``Path`` with the wrong host as much
+        as possible.
+        """
+        if self.host != host:
+            raise Exception(f"This path ({self}) is not for the host {host!r}")
+        return str(self._path)
+
+    # }}}
+
+    # PurePosixPath like API {{{
+    # Mimick the API of pathlib's PurePosixPath as much as possible.  This
+    # allows users to use tbot's Path just like the one they know from pathlib.
+    def _prepare_args_list(self, args: Iterable[Any]) -> List[Any]:
+        ret = list()
+        for arg in args:
+            if isinstance(arg, Path):
+                assert arg.host == self.host
+                ret.append(arg._path)
+            else:
+                ret.append(arg)
+        return ret
+
+    def __eq__(self, other: Any) -> bool:
+        if not isinstance(other, Path):
+            return NotImplemented
+        return self._host == other._host and self._path == other._path
+
+    def __hash__(self) -> int:
+        return hash((self._host, self._path))
+
+    def __lt__(self, other: Any) -> bool:
+        if not isinstance(other, Path):
+            return NotImplemented
+        return self._path < other._path
+
+    def __le__(self, other: Any) -> bool:
+        if not isinstance(other, Path):
+            return NotImplemented
+        return self._path <= other._path
+
+    def __gt__(self, other: Any) -> bool:
+        if not isinstance(other, Path):
+            return NotImplemented
+        return self._path > other._path
+
+    def __ge__(self, other: Any) -> bool:
+        if not isinstance(other, Path):
+            return NotImplemented
+        return self._path >= other._path
+
+    @property
+    def name(self) -> str:
+        return self._path.name
+
+    @property
+    def suffix(self) -> str:
+        return self._path.suffix
+
+    @property
+    def suffixes(self) -> List[str]:
+        return self._path.suffixes
+
+    @property
+    def stem(self) -> str:
+        return self._path.stem
+
+    def with_name(self, name: str) -> "Path[H]":
+        return Path(self._host, self._path.with_name(name))
+
+    def with_stem(self, stem: str) -> "Path[H]":
+        return Path(self._host, self._path.with_name(stem + self._path.suffix))
+
+    def with_suffix(self, suffix: str) -> "Path[H]":
+        return Path(self._host, self._path.with_suffix(suffix))
+
+    def relative_to(self, *other: Any) -> "Path[H]":
+        return Path(self._host, self._path.relative_to(*self._prepare_args_list(other)))
+
+    def is_relative_to(self, *other: Any) -> bool:
```

Comment:
Two tbot Paths associated with different hosts shall not be relative to one another even if their paths are. This should definitvely enforce other to be a tbot Path, IMHO.

Reply:
To keep consistent with `relative_to()` and pathlib, I'd like to leave it like it is:  If a tbot-path is passed in it will be checked, but we don't enforce using one to allow e.g. writing simple checks without too much clutter:

```python
somefile = ...
if somefile.is_relative_to("/tmp"):
    print("tmp file")
```

Even if someone were to run a check against a wrong path, this can't do much harm I think, as they'll get an error later when actually trying to use their path with the wrong host.